### PR TITLE
gateway2: enable self-managed Gateways

### PR DESCRIPTION
**Pull request by shashankram**
# Description

Adds capability to integrate self-managed gateways that are 
not meant to be auto-provisioned by the
controller. This is required to support use cases
where gateways are statically provisioned, such
as when running the gateways external to k8s on
VMs.

It adds a selfManaged field to the GatewayParameters CRD which 
is used to skip the deployment of Gateway related objects (deployment, service, etc.). 
When a gateway is self-managed, it is expected to be
correctly bootstrapped with an envoy config that
matches the Gateway resource's name and namespace, 
specifically the `node` field must be configured
correctly to link a self-managed gateway to the
Gateway resource.

## API changes
Adds `selfManaged` field to GatewayParameters

## Interesting decisions
Standardized behavior to configure Gateway parameters.

## Testing steps

E2e test

# Checklist:

- [X] I have performed a self-review of my own code
- [X] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation TODO
- [X] I have added tests that prove my fix is effective or that my feature works

### CODE REVIEW

**Review by ilackarms — APPROVED** [review]

File: projects/gateway2/api/v1alpha1/gateway_parameters.proto [hunk 1]
```diff
@@ -24,6 +25,9 @@ message GatewayParametersSpec {
   oneof environment_type {
     // The proxy will be deployed on Kubernetes.
     KubernetesProxyConfig kube = 1;
+
+    // The proxy will be self-managed and not auto-provisioned.
+    google.protobuf.Struct self_managed = 2;
```
Comment (yuval-k) [inline projects/gateway2/api/v1alpha1/gateway_parameters.proto:30]: why is this a Struct? did you mean to use Empty?

Comment (shashankram) [reply projects/gateway2/api/v1alpha1/gateway_parameters.proto:30]: Yes, per discussion with @ilackarms it's a convention we use for oneof instead of setting a bool

**Review by ilackarms — DISMISSED** [review]

File: projects/gateway2/api/v1alpha1/gateway_parameters.proto [hunk 2]
```diff
@@ -24,6 +25,9 @@ message GatewayParametersSpec {
   oneof environment_type {
     // The proxy will be deployed on Kubernetes.
     KubernetesProxyConfig kube = 1;
+
+    // The proxy will be self-managed and not auto-provisioned.
+    google.protobuf.Empty self_managed = 2;
```
Comment (lgadban) [inline projects/gateway2/api/v1alpha1/gateway_parameters.proto:30]: quick drive-by: do we not want to reference static provisioning in the naming here?
"Self managed" does sound completely reasonable but is "statically provisioned" the more accepted term?

Comment (shashankram) [reply projects/gateway2/api/v1alpha1/gateway_parameters.proto:30]: My personal feeling is that `static` is ambiguous, and `selfManaged` is clearer.

File: projects/gateway2/api/v1alpha1/gateway_parameters.proto (hunk 1 above)
Comment (EItanya) [reply projects/gateway2/api/v1alpha1/gateway_parameters.proto]: I agree with @yuval-k  that we usually use `google.protobuf.Empty` since there's no content

**Review by yuval-k — APPROVED** [review]
